examples/vision/video_classification.py

```diff
@@ -0,0 +1,466 @@
+    )
+    feature_extractor.trainable = False
+
+    inputs = keras.Input((IMG_SIZE, IMG_SIZE, 3))
+    preprocessed = keras.applications.inception_v3.preprocess_input(inputs)
+
+    outputs = feature_extractor(preprocessed)
+    return keras.Model(inputs, outputs, name="feature_extractor")
+
+
+"""
+We now create TensorFlow dataset objects from the video frame paths to efficiently
+extract their features.
+"""
+
+
+def load_image(image_path):
+    image = tf.io.read_file(image_path)
+    image = tf.image.decode_jpeg(image, 3)
+    image = tf.image.resize(image, (IMG_SIZE, IMG_SIZE))
+    return image
+
+
+train_images_list = new_train_df["image_path"].values.tolist()
+train_ds = tf.data.Dataset.from_tensor_slices(train_images_list)
+train_ds = train_ds.map(
+    load_image, num_parallel_calls=tf.data.AUTOTUNE, deterministic=True
+).batch(BATCH_SIZE)
+
+test_images_list = new_test_df["image_path"].values.tolist()
+test_ds = tf.data.Dataset.from_tensor_slices(test_images_list)
+test_ds = test_ds.map(
+    load_image, num_parallel_calls=tf.data.AUTOTUNE, deterministic=True
+).batch(BATCH_SIZE)
+
+feature_extractor = build_feature_extractor()
+train_features = feature_extractor.predict(train_ds)
+test_features = feature_extractor.predict(test_ds)
+
+"""
+## Reshape features and prepare labels
+
+Now that we have extracted the frame features, we need to feed them to our recurrent
+network considering the temporality present inside of them. Data fed to the recurrent
+layers must have a time-step dimension. So, how do we incorporate that in the extracted
+features?
+
+The features have a shape of `(len(train_frames), 5, 5, 2048)`. Recall that each video is
+now represented as an ordered sequence of **five** frames. So, we can reshape our features
+like so: `(len(train_frames) / 5, 5, 5 * 5 * 2048)`. The second dimension (5) is now the
+time-step dimension. It turns out that this heuristic works quite well in practice. It
+just needs as adjustment in the number of examples such that it is divisible by the
+number of frames representing a video. Fortunately, for this example, we did not have to
+do that explicitly.
+"""
+
+train_features = train_features.reshape(
+    len(train_features) // SEQ_LENGTH, SEQ_LENGTH, 5 * 5 * 2048
+)
+test_features = test_features.reshape(
+    len(test_features) // SEQ_LENGTH, SEQ_LENGTH, 5 * 5 * 2048
+)
+
+"""
+We now need to prepare the labels (the action classes), which are encoded as strings.
+Neural networks do not understand string values, so they must be converted to some numerical
+form before they are fed to the model. Here we will use the
+[`StringLookup`](https://keras.io/api/layers/preprocessing_layers/categorical/string_lookup)
+layer to binarize the string labels. It will first map the strings to integer indices and then it will
+[one-hot encode](https://developers.google.com/machine-learning/glossary#one-hot-encoding)
+those integer values.
+"""
+
+nb_classes = len(np.unique(new_train_df["class"].values))
```

Comment:
Use `num_` as numbering prefix for consistency with other examples and with the API